psi-sample/common/build.gradle

```diff
@@ -0,0 +1,16 @@
+plugins {
+    id 'java'
+}
+
+version 'unspecified'
+
+sourceCompatibility = 1.8
```

Comment:
sourceCompatibility Java version compatibility to use when compiling Java source. 

targetCompatibility is the Java version to generate classes for. The default value for targetCompatibility  is sourceCompatibility and hence that's why we dont need to specify targetCompatibility.
If you do not specify sourceCompatibility I get below error

No matching variant of project :common was found. The consumer was configured to find a runtime of a library compatible with Java 8, packaged as a jar, and its dependencies declared externally but:
          - Variant 'apiElements' capability overlay:common:unspecified declares a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares an API of a component compatible with Java 11 and the consumer needed a runtime of a component compatible with Java 8
          - Variant 'runtimeElements' capability overlay:common:unspecified declares a runtime of a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares a component compatible with Java 11 and the consumer needed a component compatible with Java 8